config/routes.rb

```diff
@@ -1,10 +1,20 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # All users page
+  get '/users', to: 'users#index', as: :users
+
+  # Single user page
+  get '/users/:id', to: 'users#show', as: :user
+
+  # All posts by a given user page
+  get '/users/:user_id/posts', to: 'posts#index', as: :user_posts
+
+  # Single post page
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
```

Comment:
- [x] [optional] @DevKaranJ It would have been nice if you used resources to create your routes, it's a tool that helps us write routes code faster, its easier to read and also scale in the future.
```suggestion
     resources :users, only: [:index, :show] do
         resources :posts, only: [:index, :show]
     end
```